### DISCUSSION TITLE
Internal error server

### DISCUSSION THREAD

**Discussion by gbertolottiKS** [nextcloud/all-in-one#3056] [Questions] [answered]
### Steps to reproduce
1. Use Nextcloud AIO

### Expected behavior 
Nothing bad happen

### Actual behavior 
Suddenly all the instance crash, and a black window is showed with the text "Internal error server" and some other things

### Host OS 
Ubuntu 22.04

#### Nextcloud AIO version 
26.0.3

#### Current channel 
latest (6.3.0)

#### Other valuable info 
It happened twice in these last two days.
To solve it, we only had to reboot the server.

Below some logs, as attachments:
- Database
- Nextcloud

[Logs.zip](https://github.com/nextcloud/all-in-one/files/12160613/Logs.zip)

Tell us if more are needed.

**Comment by szaimen** [comment, marked as answer]
Hi, can you set this in your config.php, restart your server and check if this resolves the problems?
https://github.com/nextcloud/all-in-one/blob/dace781b4ee2d7a1f5df15c63d7934f1bd5b0fc6/Containers/nextcloud/config/postgresql.php#L6

**Reply by LedyBacer** [reply]
This solved the problem, thanks

**Reply by gbertolottiKS** [reply]
Hi,
just to follow up: we implemented what you suggested, and it seems to work!
In case of it could be useful, we attach our system info.
[2023-08-03.md](https://github.com/nextcloud/all-in-one/files/12250297/2023-08-03.md)

However, ours feeling is that something (when the crashes occurred) triggers this spam of huge amount of connections to the DB. Could this be something to investigate?

Thanks a lot

**Reply by szaimen** [reply]
This is now released with v7.0.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by gbertolottiKS** [comment]
Thanks, we'll try and let you know.

Just to know: why this configuration isn't enabled by default? Do you think that this could solve the problem of the connection spam?

Thanks

**Comment by szaimen** [comment]
Because we didnt know existed until recently

**Comment by szaimen** [comment]
Also not completely sure if it works as intended

**Comment by peterpicard** [comment]
I tried this and it didn't work, as it still produced FATAL errors of "FATAL:  sorry, too many clients already" and "FATAL:  remaining connection slots are reserved for non-replication superuser connections" after trying to browse the photos web page. Anything else I can try?

Sadly, I tried rolling back to 26.0.0 but then other things don't work, as it seems the new version (27.0.1 ) migrated some configuration and files in S3 after it started for the first time. Now I'm stuck with a broken nextcloud server with thousands of photos and documents that I can't really use or depend on. I guess next time I'll have to do a proper test of the new version before jumping in.

Please help with a fix. I'm happy to try out any suggestions.

Thank you.

**Reply by szaimen** [reply]
Restoring a backup should fix this

**Reply by peterpicard** [reply]
The EC2 instance was snapshotted before the update and after. I rolled back to the snapshot pre-update and that didn't work well anymore. Seems changes were made in the S3 storage when the new version was up and running. I also remember running an indexing command as per the instructions in the UI, which seems to have created a mismatch between the DB state and S3 storage state. I didn't do a backup and rollback of my multi-terrabyte S3 bucket (wrongly) thinking it wasn't needed.

Anyway, I fixed this issue by editing the "/var/lib/docker/volumes/nextcloud_db/_data/postgresql.conf" file in the "nextcloud_db" docker volume and changing the "max_connection" value from 100 to 1000. Now it seems there are enough connections to not cause FATAL errors. This fix is permanent, as long as you don't delete and recreate the volume ;)
